server/src/main/java/org/apache/cassandra/sidecar/utils/InstanceMetadataFetcher.java

```diff
@@ -77,60 +80,40 @@ public InstanceMetadata instance(int instanceId) throws NoSuchCassandraInstanceE
     }
 
     /**
-     * Returns the {@link CassandraAdapterDelegate} for the given {@code host}. When the {@code host} is {@code null},
-     * returns the delegate for the first instance from the list of configured instances.
+     * Iterate through the local instances and call the function on the first available instance
      *
-     * @param host the Cassandra instance hostname or IP address
-     * @return the {@link CassandraAdapterDelegate} for the given {@code host}, or the first instance when {@code host}
-     * is {@code null}
-     * @throws NoSuchCassandraInstanceException when the Cassandra instance with {@code host} does not exist
-     * @throws CassandraUnavailableException  when Cassandra is not yet connected
+     * @param function function applies to {@link CassandraAdapterDelegate}
+     * @return function eval result. Null can be returned when all local instances are exhausted
+     * @param <T> type of the result
+     * @throws CassandraUnavailableException when all local instances are exhausted.
      */
     @NotNull
-    public CassandraAdapterDelegate delegate(@Nullable String host) throws NoSuchCassandraInstanceException, CassandraUnavailableException
```

Comment:
:(

Reply:
+1 since we seem to access `CassandraAdaptorDelegate` in many handlers shall we keep `delegate()`?

Reply:
ok. I will add this specific method back, since both of you want it back. 

In my opinion, retrieving delegate is not the responsibility of `InstanceMetadataFetcher`. It does not add a lot of character to retrieve from instanceMetadata. However, the methods have been there before this patch, probably could be exempted.